src/pages/edge/adobe-journey-optimizer/code-based/tabs/tutorial.md

```diff
@@ -0,0 +1,179 @@
+<Variant platform="android" function="updatepropositionsforsurfaces" repeat="2"/>
+
+#### Java
+
+```java
+final Surface surface1 = new Surface("mainActivity#banner");
+final Surface surface2 = new Surface("secondActivity#promotions");
+ 
+final List<Surface> surfaces = new ArrayList<>();
+surfaces.add(surface1);
+surfaces.add(surface2);
+ 
+// fetch propositions from server and cache in-memory
+Messaging.updatePropositionsForSurfaces(surfaces)
+```
+
+<Variant platform="ios" function="updatepropositionsforsurfaces" repeat="2"/>
+
+#### Swift
+
+```swift
+let surface1 = Surface("mainActivity#banner")
+let surface2 = Surface("secondActivity#promotions")
+ 
+/// fetch propositions from server and cache in-memory
+Messaging.updatePropositionsForSurfaces([surface1, surface2])
+```
+
+<Variant platform="android" function="getpropositionsforsurfaces" repeat="2"/>
+
+#### Java
+
+```java
+final Surface surface1 = new Surface("mainActivity#banner");
+final Surface surface2 = new Surface("secondActivity#promotions");
+ 
+final List<Surface> surfaces = new ArrayList<>();
+surfaces.add(surface1);
+surfaces.add(surface2);
+ 
+Messaging.getPropositionsForSurfaces(surfaces, new AdobeCallbackWithError<Map<Surface, List<Proposition>>>() {
+    @Override
+    public void fail(final AdobeError adobeError) {
+        // handle error
+    }
+ 
+    @Override
+    public void call(Map<Surface, List<Proposition>> propositionsMap) {
+        if (propositionsMap != null && !propositionsMap.isEmpty()) {
+           // get the content for the given surfaces
+        }
+    }
+});
+```
+
+<Variant platform="ios" function="getpropositionsforsurfaces" repeat="2"/>
+
+#### Swift
+
+```swift
+let surface1 = Surface(path: "myView#button")
+let surface2 = Surface(path: "myViewAttributes")
+ 
+Messaging.getPropositionsForSurfaces([surface1, surface2]) { propositionsDict, error in
+    guard error == nil else {
+        /// handle error
+        return
+    }
+ 
+    guard let propositionsDict = propositionsDict else {
+        /// bail early if no propositions
+        return
+    }
+     
+    /// get the content for the given surfaces
+    if let propositionsForSurface1 = propositionsDict[surface1] {
+        /// read surface1 propositions
+    }
+ 
+    if let propositionsForSurface2 = propositionsDict[surface2] {
+        /// read surface2 propositions
+    }
+}
+```
+
+<Variant platform="android" function="using-propositions" repeat="2"/>
+
+#### Java
+
+```java
+// get the propositions for surface1
+if (propositionsMap.contains(surface1)) {
+    // read propositions for surface1
+    final List<Proposition> propositionsForSurface1 = propositionsMap.get(surface1)
+    // iterate through items in proposition
+    for (final PropositionItem propositionItem: propositionsForSurface1.get(0).getItems()) {
```

Comment:
we should also check to make sure `propositionsForSurface1` contains items in our sample here...don't want NPEs in our sample code 😁